homeassistant/helpers/config_validation.py

```diff
@@ -532,6 +533,72 @@ def url(value: Any) -> str:
     raise vol.Invalid('invalid url')
 
 
+def ip_address(value: Any) -> str:
+    """Validate an IP address (IPv4/IPv6)."""
+    value = str(value)
+    try:
+        ip = ipaddress.ip_address(value)
+    except ValueError:
+        raise vol.Invalid("Invalid IP address: {}.".format(value))
+    return str(ip)
+
+
+def ipv4_address(value: Any) -> str:
```

Comment:
I think you can merge this and `ipv6_address` into the single `ip_address` validator and extend it to accept an optional parameter to specify the IP version.  Take a look at the `deprecated` validator as an example of a validator with parameters.

Reply:
Okay. Should I merge the MAC address validation functions as well?

Reply:
I'm sorry, I didn't find an easy way to pass extra arguments to the function and receive user input data at the same time using voluptuous.Schema and Homeassistant. Note that in the example that you gave me the function cv.deprecated() does not receive any input data from the user. It's not called by voluptuous.Schema and all the inputs are hard-coded, none comes from configuration file. I couldn't find any example code or documentation for this, and as the extra arguments was a must to merge the functions, I will not be able to do it for now.